### PR TITLE
Notif gae

### PR DESCRIPTION
**Pull request by ethannchen**
1. switched back to token schema v1 (one token one doc)
2. replaced python multiprocessing with Expo Server SDK method push_client.publish_multiple
3. tested against remote server dev app engine
4. fixed "parent" field issue

### CODE REVIEW

File: firestore/incidents.py [hunk 1]
```diff
@@ -18,37 +18,45 @@ class Incident(mdl.Model):
...
+        required=True, default={"twitter": None, "linkedin": None}
+    )
+    donation_link = mdl.TextField()  #  Donation: link to donation website
+    police_tip_line = (
+        mdl.TextField()
+    )  # Police Tip Line: phone number to provide tips to police to help capture the suspect
+    help_the_victim = (
+        mdl.TextField()
+    )  # Help the victim: other free style text about how people can help the victim
+    parent_doc = mdl.TextField(column_name="parent")
```
Comment (lima01) [inline firestore/incidents.py:31]: Are you sure this will work?

File: main.py [hunk 2]
```diff
@@ -27,19 +27,23 @@
 import firestore.admins
 from common import User
 from firestore.incidents import deleteIncident, getIncidents, getStats, insertIncident
-from firestore.tokens import registerNewToken
+from firestore.tokens import add_token
 import incident_publisher
 
-from flask_limiter import Limiter
-from flask_limiter.util import get_remote_address
+# import fireo
```
Comment (lima01) [inline main.py:33]: can you clean this code?

File: main.py [hunk 3]
```diff
@@ -27,19 +27,23 @@
...
+# from flask_limiter.util import get_remote_address
 
 # [END gae_python3_datastore_store_and_fetch_user_times]
 # [END gae_python38_datastore_store_and_fetch_user_times]
 app = Flask(__name__)
 # cors = CORS(app, resources={r"/*": {"origins": "*"}})
 cors = CORS(app)
 firebase_request_adapter = requests.Request()
-limiter = Limiter(get_remote_address, app=app, default_limits=["200/day", "50/hour"])
+# limiter = Limiter(get_remote_address, app=app, default_limits=["200/day", "50/hour"])
```
Comment (lima01) [inline main.py:46]: pls clean up unused code.

File: main.py [hunk 4]
```diff
@@ -209,16 +213,28 @@ def publish_incidents():
...
     deviceId = request.get_json().get("deviceId", None)
     token = request.get_json().get("token", None)
+    if not deviceId:
+        raise ValueError("No deviceId detected")
     if not token:
-        raise ValueError("no token detected ")
-    res = registerNewToken(deviceId, token)
+        raise ValueError("No token detected")
+
+    token_q.put(deviceId)
```
Comment (lima01) [inline main.py:230]: why use a queue here?
what does it do?

File: main.py [hunk 5]
```diff
@@ -209,16 +213,28 @@ def publish_incidents():
...
+    while not token_q.empty() and token_q.queue[0] != deviceId:
```
Comment (lima01) [inline main.py:231]: what does this loop do? could it cause dead loop?

**Review by lima01 — CHANGES_REQUESTED** [review]

File: firestore/incidents.py (hunk 1 above)
Comment (ethannchen) [reply firestore/incidents.py:31]: Ref: https://github.com/octabytes/FireO/issues/20#issuecomment-556989359

File: main.py (hunk 4 above)
Comment (ethannchen) [reply main.py:230]: they are unused old code, i will remove that

File: main.py (hunk 5 above)
Comment (ethannchen) [reply main.py]: they are unused old code, i will remove that

File: firestore/tokens_v2.py
Comment (lima01) [inline firestore/tokens_v2.py:1]: Shall we delete this  file?

File: firestore/tokens_v3.py
Comment (lima01) [inline firestore/tokens_v3.py:1]: Shall we delete this  file?

File: firestore/tokens_v3_0.py
Comment (lima01) [inline firestore/tokens_v3_0.py:1]: Shall we delete this  file?

**Review by lima01 — APPROVED** [review]

**Pull request merged**